platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/extensions/BusyTime.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.metrics.extensions;
+
+import com.swirlds.common.metrics.FloatFormats;
+import com.swirlds.common.metrics.FunctionGauge;
+import com.swirlds.common.metrics.Metrics;
+import com.swirlds.common.time.IntegerEpochTime;
+import com.swirlds.common.time.OSTime;
+import com.swirlds.common.time.Time;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * A metric that measures the fraction of time that a thread is busy. This metric could be used to track the overall busy
+ * time of a thread, or the busy time of a specific subtask. The granularity of this metric is in microseconds. A
+ * snapshot of this metric must be taken at least once every 34 minutes in order to get accurate data.
+ */
+public class BusyTime {
+    /** passed to the accumulator method to indicate that work has started */
+    private static final long WORK_START = 0;
+    /** passed to the accumulator method to indicate that work has ended */
+    private static final long WORK_END = 1;
+    /** the initial value of status when the instance is created */
+    private static final int INITIAL_STATUS = -1;
+    /** An instance that provides the current time */
+    private final IntegerEpochTime time;
+    /** Used to atomically update and reset the time and status */
+    private final AtomicLong accumulator;
+
+    /**
+     * The default constructor, uses the {@link OSTime} instance to get the current time
+     */
+    public BusyTime() {
+        this(OSTime.getInstance());
+    }
+
+    /**
+     * A constructor where a custom {@link Time} instance could be supplied
+     *
+     * @param time
+     * 		provides the current time
+     */
+    public BusyTime(@NonNull final Time time) {
+        this.time = new IntegerEpochTime(time);
+        this.accumulator = new AtomicLong(IntPairUtils.combine(this.time.getMicroTime(), INITIAL_STATUS));
+    }
+
+    /**
+     * Adds a {@link FunctionGauge} to the given {@link Metrics} instance
+     *
+     * @param metrics
+     * 		the metrics instance to add the metric to
+     * @param config
+     * 		the configuration for this metric
+     */
+    public void addMetric(@NonNull final Metrics metrics, @NonNull final DefaultMetricConfig config) {
```

Comment:
It would be possible to do it in the constructor, but I have a few reasons why I made it separate:
- Unit testing is simpler
- I have a future use case in mind where you would want the instance but don't want to add it as a metric
- This makes more sense and is more flexible, a constructor constructs, and another methods can add it to metrics